bottlecap/src/lifecycle/invocation/triggers/lambda_function_url_event.rs

```diff
@@ -0,0 +1,309 @@
+
+        let http_url = format!(
+            "https://{domain_name}{path}",
+            domain_name = self.request_context.domain_name.clone(),
+            path = self.request_context.http.path.clone()
+        );
+
+        let start_time = (self.request_context.time_epoch as f64 * MS_TO_NS) as i64;
+        // todo: service mapping and peer service
+        let service_name = self.request_context.domain_name.clone();
+
+        span.name = String::from("aws.lambda.url");
+        span.service = service_name;
+        span.resource = resource;
+        span.r#type = String::from("http");
+        span.start = start_time;
+        span.meta.extend([
+            (
+                "endpoint".to_string(),
+                self.request_context.http.path.clone(),
+            ),
+            ("http.url".to_string(), http_url),
+            (
+                "http.method".to_string(),
+                self.request_context.http.method.clone(),
+            ),
+            (
+                "http.user_agent".to_string(),
+                self.request_context.http.user_agent.clone(),
+            ),
+            (
+                "http.source_ip".to_string(),
+                self.request_context.http.source_ip.clone(),
+            ),
+            (
+                "http.protocol".to_string(),
+                self.request_context.http.protocol.clone(),
+            ),
+            ("operation_name".to_string(), "aws.lambda.url".to_string()),
+            (
+                "request_id".to_string(),
+                self.request_context.request_id.clone(),
+            ),
+        ]);
+    }
+
+    fn get_tags(&self) -> HashMap<String, String> {
+        let mut tags = HashMap::from([
+            (
+                "http.url".to_string(),
+                format!(
+                    "https://{domain_name}{path}",
+                    domain_name = self.request_context.domain_name.clone(),
+                    path = self.request_context.http.path.clone()
+                ),
+            ),
+            // path and URL are full
+            // /users/12345/profile
+            (
+                "http.url_details.path".to_string(),
+                self.request_context.http.path.clone(),
+            ),
+            (
+                "http.method".to_string(),
+                self.request_context.http.method.clone(),
+            ),
+            (
+                "http.user_agent".to_string(),
+                self.request_context.http.user_agent.clone(),
+            ),
+            (
+                FUNCTION_TRIGGER_EVENT_SOURCE_TAG.to_string(),
+                "lambda-function-url".to_string(),
+            ),
+        ]);
+
+        if let Some(referer) = self.headers.get("referer") {
+            tags.insert("http.referer".to_string(), referer.clone());
+        }
+
+        tags
+    }
+
+    fn get_arn(&self, region: &str) -> String {
+        let function_name = env::var("AWS_LAMBDA_FUNCTION_NAME").unwrap_or_default();
```

Comment:
Don't like doing this, but I'd rather do this than pass another field and modify the whole struct trigger architecture